glean_parser/templates/ruby_server.jinja2

```diff
@@ -0,0 +1,138 @@
+{# The final Ruby code is autogenerated, but this
+Jinja2 template is not. Please file bugs! #}
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# AUTOGENERATED BY glean_parser v{{ parser_version }}. DO NOT EDIT.
+
+# frozen_string_literal: true
+
+# requires json, securerandom, and logger libs
+require 'json'
+require 'securerandom'
+require 'logger'
+require 'rbconfig'
+
+
+# this will be used for identifying logs that need to forward to Moz Data Pipeline
+GLEAN_EVENT_MOZLOG_TYPE = 'glean-server-event'
+
+module OS
+  def self.name
+    case RbConfig::CONFIG['host_os']
+    
+    when /linux/
+      'Linux'
+    when /darwin/
+      'OS X'
+    when /mswin|mingw32|windows/
+      'Windows'
+    when /solaris/
+      'Solaris'
+    when /bsd/
+      'BSD'
+    else
+      RbConfig::CONFIG['host_os']
+    end
+  end
+  def self.version
+    version = RbConfig::CONFIG['build_os'].match('\d{1,2}\.*\d{0,2}\.*\d{0,3}').to_s
+    if !version.nil?
+        version.to_s
+    else
+        'Unknown'
+    end
+  end
+end
+
+module GleanHelper
+  {# Create a list of ping classes. #}
+  {% for event_category, event_types in events.items() %}
+  {% for event_type, event_object in event_types.items() %}
+  class {{ (event_category + '.' + event_type) | event_class_name }}
+    # see comments for init parameter descriptions
+    def initialize(application_id:, app_display_version:, app_channel:, user_agent:, ip_address:, event_extra)
+      @application_id = application_id # string - Application Id to identify application per Glean standards
+      @app_display_version = app_display_version # string - Application Id to identify application per Glean standards
+      @app_channel = app_channel # string - Application Id to identify application per Glean standards
+      @user_agent = user_agent # string - The user agent per Glean standards
+      @ip_address = ip_address # string - The IP address that will be used to decode Geo information and scrubbed at ingestion per Glean standards
+      @event_extra = event_extra # string - JSON blob containing the backend event payload
```

Comment:
Why are we passing `event_extra` into the constructor?

actually let me pull this out into a full comment aside from the code review.

Reply:
ok